src/shared/mkdir-label.c

```
@@ -57,3 +58,11 @@ int mkdir_parents_label(const char *path, mode_t mode) {
 int mkdir_p_label(const char *path, mode_t mode) {
         return mkdir_p_internal(NULL, path, mode, UID_INVALID, UID_INVALID, 0, mkdir_label);
 }
+
+void mkdir_p_label_and_warn(const char *path, mode_t mode, const char *logsrc) {
```

this really should propagate the error. it should be up to the callers to ignore the error, not the callee. wether to warn about the failure is a different question from ignoring errors.